.github/actions/run-publish-vr-screenshot/action.yml

```diff
@@ -0,0 +1,132 @@
+    required: true
+    default: 'fluentuiv8'
+  locationPostfix:
+    description: 'Location postfix'
+    required: true
+    default: 'vrscreenshotv8'
+  clientName:
+    description: 'Client name'
+    required: true
+    default: 'fluentuiv8'
+
+runs:
+  using: 'composite'
+  steps:
+    - uses: nrwl/nx-set-shas@v4
+      with:
+        main-branch-name: 'master'
+
+    - uses: actions/setup-node@v4
+      with:
+        cache: 'yarn'
+        node-version: '20'
+
+    - run: yarn install --frozen-lockfile
+      shell: bash
+
+    - run: yarn playwright install --with-deps
+      shell: bash
+
+    - name: Check if vrTests should be skipped
+      shell: bash
+      run: |
+        isPR=${{ github.event_name == 'pull_request' }}
+        echo "isPR=$isPR" >> $GITHUB_ENV
+        if [[ $isPR == true ]]; then
+          packageAffected=$(yarn --silent check:affected --project ${{ inputs.vrTestPackageName }})
+          if [[ $packageAffected == false ]]; then
+            echo "In PR pipeline but NOT affecting test package. Skipping test run"
+            echo "vrTestSkip=yes" >> $GITHUB_ENV
+          else
+            echo "In PR pipeline and affecting test package. NOT Skipping test run"
+            echo "vrTestSkip=no" >> $GITHUB_ENV
+          fi
+        else
+          echo "Not in PR pipeline. NOT Skipping test run"
+          echo "vrTestSkip=no" >> $GITHUB_ENV
+        fi
+
+    - name: Run VR tests
+      if: ${{ env.vrTestSkip == 'no' }}
+      shell: bash
+      run: yarn nx run ${{ inputs.vrTestPackageName }}:test-vr --nxBail
+
+    - name: Collate Artifacts
+      if: ${{ env.vrTestSkip == 'no' }}
+      shell: bash
+      run: |
+        mkdir -p screenshots
+        cp -rf ${{ inputs.vrTestPackagePath }}/dist/screenshots/*.png screenshots/
+
+    - name: Upload VR screenshots
+      if: ${{ env.vrTestSkip == 'no' }}
+      uses: actions/upload-artifact@v3
+      with:
+        name: vrscreenshot${{ inputs.fluentVersion }}
+        path: screenshots
+
+    - name: VR App - Create Policy
+      if: ${{ env.isPR == 'true' }}
+      shell: bash
+      run: |
+        set -exuo pipefail
+        npx vr-approval-cli@0.4.11 create-policy --nonBlockingPipelines '{"${{ env.pipelineId }}":{"pipelineStatus": "PENDING","pipelineName": "${{ env.pipelineName }}"}}' --clientType 'FLUENTUI'
+      # env:
+      # VR_APP_API_URL: ${{ secrets.VR_APP_API_URL }}
+      # TENANT_ID: ${{ secrets.TenantId }}
+      # PRINCIPAL_CLIENT_ID: ${{ secrets.PrincipalClientId }}
+      # SERVICE_CONNECTION_ID: ${{ secrets.ServiceConnectionId }}
+      # GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+    - name: Run screenshotdiff
+      if: ${{ env.isPR == 'true' && env.vrTestSkip == 'no' }}
+      uses: azure/cli@v2
+      env:
+        # ciDefinitionId is set to 205 because that is the ID of the baseline pipeline (https://uifabric.visualstudio.com/fabricpublic/_build?definitionId=205) used by the master branch
+        CI_DEFINITION_ID: 205
```

Comment:
im not sure how this would be actually used in GHA context

this is what is availabe:

<img width="780" alt="image" src="https://github.com/user-attachments/assets/98470d5e-cd44-4c79-afec-3c1f12a77348">

**Note:**  I didn't create separate workflow for baseline as we had on ADO (https://github.com/microsoft/fluentui/blob/master/azure-pipelines.vrt-baseline.yml) to mitigate duplication.